src/viability_phase.py

```diff
@@ -17,19 +90,49 @@ def viability_phase(learning_profiles: list, num_iterations: int = -1,
         learning_profiles (list): A list of all Learning Profiles to train.
         num_iterations (int): Number of evaluation iterations. Set to -1 if
             all training songs should be depleted. Default is -1.
-        seed_percent (float): Percent of data to be used as seed. 
+        seed_percent (float): Percent of data to be used as seed.
             Default is 0.1.
+        n_threads (int): The number of thread to use for evaluation.
+            Default is 1.
     """
 
-    # Perform viability test for all learning profiles
-    for lp in learning_profiles:
-        print(f"Evaluating {lp}")
-        MSE_arousal, MSE_valence = _evaluate(lp, num_iterations, seed_percent)
-        lp.set_MSE(MSE_arousal, MSE_valence)
+    # Print thread configuration
+    print(
+        f"Using {n_threads} threads for "
+        f"{len(learning_profiles)} learning profiles."
+    )
+
+    threads = []
+
+    # Start threads
+    for i, lps in enumerate(_chunk_it(learning_profiles, n_threads)):
+
+        # Print chunk info
+        tqdm.get_lock().acquire()
 
-        print(f"Done!")
-        print(f"Arousal MSE per iteration: {lp.get_MSE_arousal()}")
-        print(f"Valence MSE per iteration: {lp.get_MSE_valence()}")
+        tqdm.write(
+            f"Starting evaluation of {len(lps)} "
+            f"learning profiles on thread {i}:"
+        )
+
+        tqdm.write("{")
+        for lp in lps:
+            tqdm.write(f"\t{str(lp)} | ID: {lp.get_id()}")
+        tqdm.write("}")
+
+        tqdm.get_lock().release()
+
+        # Create and start thread
+        th = Thread(target=_evaluate_learning_profiles,
+                    args=[i, lps, num_iterations, seed_percent])
+        th.start()
+        threads.append(th)
+
+    # Wait for threads to exit
+    for th in threads:
+        th.join()
+
+    tqdm.write("All threads finished successfully!")
 
         yield lp
```

Comment:
How did the code run with indentation error? 

The yielding of each profile after they have completed does not work now.
This was implemented like this bc `evaluation_phase` can save (to disk) the profile after it has completed, and not wait for all profiles to finish before saving.

Reply:
Definitely an error, good catch!